docs/pages/toolchain/katana/interact.md

```diff
@@ -0,0 +1,377 @@
+            }
+
+            self.emit(VoteCast { voter: caller, vote: vote, });
+        }
+    }
+
+    /// @dev Internal Functions implementation for the Vote contract
+    #[generate_trait]
+    impl InternalFunctions of InternalFunctionsTrait {
+        /// @dev Registers the voters and initializes their voting status to true (can vote)
+        fn _register_voters(
+            ref self: ContractState,
+            voter_1: ContractAddress,
+            voter_2: ContractAddress,
+            voter_3: ContractAddress
+        ) {
+            self.registered_voter.write(voter_1, true);
+            self.can_vote.write(voter_1, true);
+
+            self.registered_voter.write(voter_2, true);
+            self.can_vote.write(voter_2, true);
+
+            self.registered_voter.write(voter_3, true);
+            self.can_vote.write(voter_3, true);
+        }
+    }
+
+    /// @dev Asserts implementation for the Vote contract
+    #[generate_trait]
+    impl AssertsImpl of AssertsTrait {
+        // @dev Internal function that checks if an address is allowed to vote
+        fn _assert_allowed(ref self: ContractState, address: ContractAddress) {
+            let is_voter: bool = self.registered_voter.read((address));
+            let can_vote: bool = self.can_vote.read((address));
+
+            if (can_vote == false) {
+                self.emit(UnauthorizedAttempt { unauthorized_address: address, });
+            }
+
+            assert(is_voter == true, 'USER_NOT_REGISTERED');
+            assert(can_vote == true, 'USER_ALREADY_VOTED');
+        }
+    }
+
+    /// @dev Implement the VotingResultTrait for the Vote contract
+    #[generate_trait]
+    impl VoteResultFunctionsImpl of VoteResultFunctionsTrait {
+        // @dev Internal function to get the voting results (yes and no vote counts)
+        fn _get_voting_result(self: @ContractState) -> (u8, u8) {
+            let n_yes: u8 = self.yes_votes.read();
+            let n_no: u8 = self.no_votes.read();
+
+            (n_yes, n_no)
+        }
+
+        // @dev Internal function to calculate the voting results in percentage
+        fn _get_voting_result_in_percentage(self: @ContractState) -> (u8, u8) {
+            let n_yes: u8 = self.yes_votes.read();
+            let n_no: u8 = self.no_votes.read();
+
+            let total_votes: u8 = n_yes + n_no;
+
+            if (total_votes == 0_u8) {
+                return (0, 0);
+            }
+            let yes_percentage: u8 = (n_yes * 100_u8) / (total_votes);
+            let no_percentage: u8 = (n_no * 100_u8) / (total_votes);
+
+            (yes_percentage, no_percentage)
+        }
+    }
+}
+```
+
+2. Compile contract
+
+```bash
+scarb build
+```
+
+3. Environment variables
+
+   Having compiled the smart contract, it's time to declare it with Starkli and katana. For convenient management, place the following environment variables in a .env file within the `src/` directory.
```

Comment:
We've already defined those variables at the beginning of the tutorial. Seems a better approach to actually use the `.env`. We may refer to the starkli book for the first part then, and keep this one as it's more intuitive and re-usable?

Reply:
Yes. I was thinking how to explain well and where to keep the environment variables. I am gonna follow your suggestion on this.